showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,206 @@
+    GenericJson converted =
+        factory.fromInputStream(
+            getClass().getResourceAsStream(TEST_GDCH_CREDENTIAL_FILE), GenericJson.class);
+
+    // modify and save to a temporary folder
+    converted.set(CA_CERT_JSON_KEY, caCertPath.toAbsolutePath().toString());
+    projectId = converted.get("project").toString();
+    tokenUri = URI.create(converted.get("token_uri").toString());
+
+    File tempGdchCredentialFile = tempFolder.newFile(TEMP_CREDENTIAL_JSON_FILENAME);
+    try (FileWriter fileWriter = new FileWriter(tempGdchCredentialFile)) {
+      String preparedJson = converted.toPrettyString();
+      fileWriter.write(preparedJson);
+    }
+
+    // use temp location to instantiate credentials
+    initialCredentials = GdchCredentialsTestUtil.fromJson(converted, transportFactory);
+  }
+
+  /**
+   * {@link com.google.api.gax.rpc.ClientContext} will create a new {@link GdchCredentials} with an
+   * audience defaulted to the endpoint if the audience is not manually passed. This test confirms
+   * that a new credential is created from the context and can be refreshed
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testCreateClient_withGdchCredentialAndNoAudience_defaultsToEndpointBasedAudience()
+      throws IOException {
+
+    // we create the client as usual - no audience passed
+    String testEndpoint = "custom-endpoint:123";
+    settings = settings.toBuilder().setEndpoint(testEndpoint).build();
+    context = ClientContext.create(settings);
+    stubSettings = EchoStubSettings.newBuilder(context).build();
+    client = EchoClient.create(stubSettings.createStub());
+
+    // We retrieve from context and from client
+    // the client has only access to creds provider, which may differ from the actual credentials
+    // used in the Context
+    Credentials fromContext = context.getCredentials();
+    Credentials fromClient = initialCredentials;
+
+    // Since ClientContext.create() uses a modified version of GdchCredentials
+    // via GdchCredentials.createWithGdchAudience(), they should be different objects
+    assertNotSame(fromContext, fromClient);
+
+    // When credentials don't have an audience (such as the ones we passed to client creation and
+    // now stored in the
+    // provider) they will throw if we try to refresh them
+    NullPointerException expectedEx =
+        assertThrows(NullPointerException.class, () -> initialCredentials.refresh());
+    assertTrue(
+        expectedEx.getMessage().contains("Audience are not configured for GDCH service account"));
+
+    // However, the credentials prepared in ClientContext should be able to refresh since the
+    // audience would be
+    // internally defaulted the endpoint of the StubSettings
+    registerCredential(fromContext);
+    ((GdchCredentials) fromContext).refreshAccessToken();
+    String usedAudience = transportFactory.transport.getLastAudienceSent();
+    assertNotNull(usedAudience);
```

Comment:
This is redundant as we are already verifying that the audience is equal to endpoint.

Reply:
Removed